core/model/modx/processors/resource/sort.class.php

```diff
@@ -199,6 +206,17 @@ public function sortContexts() {
         return true;
     }
 
+    public function getActiveTargetMenuindex(int $target) {
+        $resource = $this->modx->getObject('modResource', $target);
```

Comment:
The `getObject` method allows you to pass in a variety of params as the second param - including raw SQL. 

While you're currently typehinting in the method, which would cause $target to always be integer, making getObject treat it as the primary key... if you change the method signature to support PHP 5.6 per my other comment, you should make sure to either cast `$target` to an int here to prevent it from being a potential SQLI vulnerability.

Reply:
Per other comment, the scalar typehint can stay, so this comment is also no longer relevant.